api/notd/collection_processor.py

```diff
@@ -83,17 +98,17 @@ async def retrieve_collection(self, address: str) -> RetrievedCollection:
             openseaCollection = {}
         retrievedCollection = RetrievedCollection(
             address=address,
-            name=collectionName or openseaCollection.get('name'),
-            symbol=collectionSymbol or openseaCollection.get('symbol'),
-            description=openseaCollection.get('description'),
-            imageUrl=openseaCollection.get('image_url'),
-            twitterUsername=openseaCollection.get('twitter_username'),
-            instagramUsername=openseaCollection.get('instagram_username'),
-            wikiUrl=openseaCollection.get('wiki_url'),
-            openseaSlug=openseaCollection.get('slug'),
-            url=openseaCollection.get('external_url'),
-            discordUrl=openseaCollection.get('discord_url'),
-            bannerImageUrl=openseaCollection.get('banner_image_url'),
+            name=collectionName or collectionMetadata['name'] or openseaCollection.get('name'),
```

Comment:
i would be very sceptical that this works - what are you testing it with? there are multiple things i doubt - collectionMetadata looks undefined if the collection doesnt have a uri (you need to initialise it to none somewhere). on top of that there is no garuntee that name is in the metadata, it should deal with the case that it's not in there.

Reply:
i would suggest adding a test_collection_processor and adding a few cases of each -
contracts that have a collectionURI
contracts that dont have an on-chain name and/or symbol
contracts that have different things set in the opensea response